src/protocol.rs

```diff
@@ -313,18 +331,22 @@ pub struct RoomInfo {
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct ConnectionRefused {
-    pub errors: Vec<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub errors: Option<Vec<String>>,
```

Comment:
```suggestion
    #[serde(default)]
    pub errors: Vec<String>,
```

Since there's no meaningful distinction between `None` and an empty `Vec`, we can get serde to collapse them into the same value and save users the hassle of unwrapping the option.